letsencrypt/client.py

```diff
@@ -293,9 +295,11 @@ def save_certificate(self, certr, chain_cert, cert_path, chain_path):
         logger.info("Server issued certificate; certificate written to %s",
                     act_cert_path)
 
+        # TODO too long, refactor... either split this into another function
```

Comment:
Yep. Should be refactored. I hate leaving around TODOs because they get neglected. What about something like:
```
def _save_chain(chain_pem, chain_path):
    """Saves chain_pem at a unique path based on chain path.

    :param str chain_pem: certificate chain in PEM format
    :param str chain_path: candidate path for the cert chain 

    :returns: absolute path to saved cert chain
    :rtype: str

    """
    chain_file, act_chain_path = le_util.unique_file(chain_path, 0o644)
    try:
        chain_file.write(chain_pem)
    finally:
        chain_file.close()

    logger.info("Cert chain written to %s", act_chain_path)

    # This expects a valid chain file
    return os.path.abspath(act_chain_path)
```

Then update save_certificate to:
```
    ...

    if chain_cert:
        chain_pem = crypto_util.dump_pyopenssl_chain(chain_cert)
        cert_chain_abspath = _save_chain(chain_pem, chain_path)
        fullchain_abspath = _save_chain(cert_pem + chain_pem, fullchain_path)

    ...

    return os.path.abspath(act_cert_path), cert_chain_abspath, fullchain_abspath
```

Note: This code is not tested